grails-app/migrations/views/stock-movement.sql

```diff
@@ -0,0 +1,131 @@
+CREATE OR REPLACE VIEW stock_movement AS
+    select
+           r.id,
+           r.name,
+           r.request_number as identifier,
+           r.description,
+           r.origin_id,
+           r.destination_id,
+           r.date_requested,
+           r.date_created,
+           r.last_updated,
+           r.requested_by_id,
+           r.created_by_id,
+           r.updated_by_id,
+           ri.item_count,
+           'OUTBOUND' as stock_movement_type,
+           s.shipment_type_id,
+           s.expected_shipping_date as date_shipped,
+           s.expected_delivery_date,
+           s.driver_name,
+           s.additional_information as comments,
+           s.id as shipment_id,
+           s.current_status as shipment_status,
+           rn.identifier as tracking_number,
+           r.id as requisition_id,
+           r.status as requisition_status,
+           r.requisition_template_id as stocklist_id,
+           null as order_id,
+           null as order_status
+    from requisition r
+        left join shipment s on s.requisition_id = r.id
+        left join (
+            select
+                   requisition_id,
+                   count(*) as item_count
+            from requisition_item
+            group by requisition_id
+            ) ri on ri.requisition_id = r.id
+
+        left join shipment_reference_number sr on s.id = sr.shipment_reference_numbers_id
+        left join reference_number rn on rn.id = sr.reference_number_id and rn.reference_number_type_id = "10"
+    where r.is_template is false
+
+union all
+
+    select
+           s.id,
+           s.name,
+           s.shipment_number as identifier,
+           s.description,
+           s.origin_id,
+           s.destination_id,
+           s.date_created as date_requested,
+           s.date_created,
+           s.last_updated,
+           s.created_by_id as requested_by_id,
+           s.created_by_id,
+           s.updated_by_id,
+           si.item_count,
+           'INBOUND' as stock_movement_type,
```

Comment:
I assume removing / refactoring this field will allow us to remove this subquery all together since all of these stock movements are already represented in the list above.

Reply:
@pmuchowski can you commend on that?

Basically what I mean is that we should just get a list of all stock movements and whether they are inbound or outbound is a determined once we know the current location. So I could pass the current location as origin or destination and that would provide enough context to determine what records should be returned.

Reply:
@jmiranda I think we can do that, but when you add ` and s.requisition_id is null` to this query and run it on obdev there will be some results, I'm not sure how those shipments were created, but you can find it in the list inbound movement page when you search by identifier